dapps/digital-twin-detail/src/components/sharing/SharingOverview.ts

```diff
@@ -0,0 +1,234 @@
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+  See the GNU Affero General Public License for more details.
+
+  You should have received a copy of the GNU Affero General Public License
+  along with this program. If not, see http://www.gnu.org/licenses/ or
+  write to the Free Software Foundation, Inc., 51 Franklin Street,
+  Fifth Floor, Boston, MA, 02110-1301 USA, or download the license from
+  the following URL: https://evan.network/license/
+*/
+
+// vue imports
+import Component, { mixins } from 'vue-class-component';
+
+// evan.network imports
+import { EvanComponent } from '@evan.network/ui-vue-core';
+import { DAppTwin, DAppContainer } from '@evan.network/digital-twin-lib';
+import { profileUtils } from '@evan.network/ui';
+
+import ShareContainerComponent from '../container/ShareContainer';
+
+/**
+ * Used to start user name / type loading directly while container resolve.
+ */
+interface LoadingSharedUserInterface {
+  containers: Array<string>;
+  name: Promise<string>;
+  type: Promise<string>;
+}
+
+/**
+ * Format of resolved "search" results
+ */
+interface SharedUserInterface {
+  address: string; // account address
+  containerNames: Array<string>; // list of containers, mapped to it's names
+  containers: Array<string>; // list of containers, for that the user is permitted;
+  icon: string; // account type material icon
+  name: string; // users name, alias or account address
+  type: string; // account type
+}
+
+interface ColumnInterface {
+  key: string;
+  label: string;
+  sortable?: boolean;
+  tdClass?: string;
+  thClass?: string;
+}
+
+@Component({ })
+export default class SharingOverview extends mixins(EvanComponent) {
+  columns: ColumnInterface[ ] = [];
+
+  data: SharedUserInterface[ ] = [];
+
+  isActiveSearch = false;
+
+  loading = true;
+
+  reverse = false;
+
+  selectedFilter = 'all';
+
+  searchTerm = '';
+
+  sortBy = 'name';
+
+  filters = [
+    { type: 'unshared', icon: 'help-circle-outline' },
+    { type: 'user', icon: 'account-outline' },
+    { type: 'company', icon: 'office-building' },
+    { type: 'all', icon: 'share-variant' },
+  ];
+
+  /**
+   * Resolve permissions for the twin and containers.
+   */
+  async created(): Promise<void> {
+    this.columns = [
+      {
+        key: 'icon',
+        label: '',
+        thClass: 'th-icon',
+      },
+      {
+        key: 'name',
+        label: this.$t('_twin-detail.sharings.table.name'),
+        sortable: true,
+        tdClass: 'truncate',
+      },
+      {
+        key: 'containers',
+        label: this.$t('_twin-detail.sharings.table.containers'),
+        tdClass: 'truncate',
+      },
+    ];
+
+    await this.resolvePermissions();
+  }
+
+  /**
+   * Load all permissions for the current opened twins.
+   */
+  async resolvePermissions(): Promise<void> {
+    const runtime = this.getRuntime();
+    const { twin: { containerAddresses } }: { twin: DAppTwin } = this.$store.state;
```

Comment:
this is really hard to read, I would recommend to use getters for vars from $store anyways.